libraries/chain/db_market.cpp

```diff
@@ -932,7 +932,9 @@ bool database::fill_settle_order( const force_settlement_object& settle, const a
 { try {
    bool filled = false;
 
-   auto issuer_fees = pay_market_fees(get(receives.asset_id), receives);
+   auto issuer_fees = ( head_block_time() < HARDFORK_CORE_1780_TIME ) ?
+      pay_market_fees(get(receives.asset_id), receives) :
+      pay_market_fees(settle.owner(*this), get(receives.asset_id), receives);
 
```

Comment:
Please add a log message if market fees are shared before the hardfork. IMO for better performance use a simple condition to decide about the log message, like `receives.options.extensions.value.reward_percent.valid()`.
Also please add a comment like `TODO: remove after HF if possible`.